File: src/Net/1. First Course/4. Third Hole/RefactoringGolf.Store/Salesman.cs
namespace RefactoringGolf.Stack
{
    public class Salesman : Employee
    {
        public int CommissionPorcentage { get; set; }

        public decimal MonthQuota { get; private set; }

        public Salesman(string firstName, string lastName, decimal fixedSalary, int commissionPorcentage)
            : base(firstName, lastName, fixedSalary)
        {
            this.CommissionPorcentage = commissionPorcentage;
        }

        public decimal NetSalary()
        {
            decimal benefits = this.MonthQuota * this.CommissionPorcentage / 100;
            decimal pensionFounds = this.FixedSalary * 10 / 100;
            decimal tax = 0;
            if (FixedSalary > 3500)
                tax = FixedSalary * 5 / 100;
            return this.FixedSalary + benefits - pensionFounds - tax;
        }

        public void UpdateMonthQuota(decimal addQuota)
        {
            MonthQuota = MonthQuota + addQuota;
        }
    }
}

File: src/Java/1. First Course/0. Initial Tee/src/main/refactoringgolf/store/Salesman.java
package refactoringgolf.store;

public class Salesman extends Employee {

	private int commissionPorcentage;
	private float monthQuota;

	public Salesman(String firstName, String lastName, float fixedSalary, int commissionPorcentage){
    	super(firstName, lastName, fixedSalary);
        this.commissionPorcentage = commissionPorcentage;
    }

	public void setCommissionPorcentage(int commissionPorcentage) {
		this.commissionPorcentage = commissionPorcentage;
	}

	public int getCommissionPorcentage() {
		return commissionPorcentage;
	}

	public float getMonthQuota() {
		return monthQuota;
	}

	public float netSalary() {
		float benefits = monthQuota * commissionPorcentage / 100;
		float pensionFounds = fixedSalary * 10 / 100;
		float tax = 0;
		if (fixedSalary > 3500)
			tax = fixedSalary * 5 / 100;
		return fixedSalary+benefits - pensionFounds - tax;
	}

	public void updateMonthQuota(float addQuota) {
		monthQuota = monthQuota + addQuota;
	}
}
